From 10c5da4e75225b579ab16e4dfc328aa7bff6ca0b Mon Sep 17 00:00:00 2001
From: Florence Blanc-Renaud <flo@redhat.com>
Date: Mon, 16 Mar 2026 12:52:27 +0100
Subject: [PATCH] IPAFileCheck: also allow 640 for kra debug log

The check ipahealthcheck.ipa.files.IPAFileCheck reports an error
with PKI 11.9.1 because this update changed the permissions on
/var/log/pki/pki-tomcat/kar/debug.$DATE.log to 0640 instead of
0644.

Allow both values as we already do for CA debug log.

Fixes: https://github.com/freeipa/freeipa-healthcheck/issues/374
Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
---
 src/ipahealthcheck/ipa/files.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/ipahealthcheck/ipa/files.py b/src/ipahealthcheck/ipa/files.py
index c4e7711..f57d9e6 100644
--- a/src/ipahealthcheck/ipa/files.py
+++ b/src/ipahealthcheck/ipa/files.py
@@ -159,7 +159,8 @@ class IPAFileCheck(IPAPlugin, FileCheck):
 
         for globpath in glob.glob("%s/debug*.log" % paths.TOMCAT_KRA_DIR):
             self.files.append(
-                (globpath, constants.PKI_USER, constants.PKI_GROUP, "0644")
+                (globpath, constants.PKI_USER, constants.PKI_GROUP,
+                 ("0644", "0640"))
             )
 
         for globpath in glob.glob(
-- 
2.52.0

